definitions/npm/styled-components_v4.x.x/flow_v0.92.x-/styled-components_v4.x.x.js

```diff
@@ -0,0 +1,410 @@
+    tfoot:                    StyledComponentType<StyledElementType<'tfoot'>, Theme>,
+    th:                       StyledComponentType<StyledElementType<'th'>, Theme>,
+    thead:                    StyledComponentType<StyledElementType<'thead'>, Theme>,
+    time:                     StyledComponentType<StyledElementType<'time'>, Theme>,
+    title:                    StyledComponentType<StyledElementType<'title'>, Theme>,
+    tr:                       StyledComponentType<StyledElementType<'tr'>, Theme>,
+    track:                    StyledComponentType<StyledElementType<'track'>, Theme>,
+    u:                        StyledComponentType<StyledElementType<'u'>, Theme>,
+    ul:                       StyledComponentType<StyledElementType<'ul'>, Theme>,
+    var:                      StyledComponentType<StyledElementType<'var'>, Theme>,
+    video:                    StyledComponentType<StyledElementType<'video'>, Theme>,
+    wbr:                      StyledComponentType<StyledElementType<'wbr'>, Theme>,
+
+    // SVG
+    circle:                   StyledComponentType<StyledElementType<'circle'>, Theme>,
+    clipPath:                 StyledComponentType<StyledElementType<'clipPath'>, Theme>,
+    defs:                     StyledComponentType<StyledElementType<'defs'>, Theme>,
+    ellipse:                  StyledComponentType<StyledElementType<'ellipse'>, Theme>,
+    g:                        StyledComponentType<StyledElementType<'g'>, Theme>,
+    image:                    StyledComponentType<StyledElementType<'image'>, Theme>,
+    line:                     StyledComponentType<StyledElementType<'line'>, Theme>,
+    linearGradient:           StyledComponentType<StyledElementType<'linearGradient'>, Theme>,
+    mask:                     StyledComponentType<StyledElementType<'mask'>, Theme>,
+    path:                     StyledComponentType<StyledElementType<'path'>, Theme>,
+    pattern:                  StyledComponentType<StyledElementType<'pattern'>, Theme>,
+    polygon:                  StyledComponentType<StyledElementType<'polygon'>, Theme>,
+    polyline:                 StyledComponentType<StyledElementType<'polyline'>, Theme>,
+    radialGradient:           StyledComponentType<StyledElementType<'radialGradient'>, Theme>,
+    rect:                     StyledComponentType<StyledElementType<'rect'>, Theme>,
+    stop:                     StyledComponentType<StyledElementType<'stop'>, Theme>,
+    svg:                      StyledComponentType<StyledElementType<'svg'>, Theme>,
+    text:                     StyledComponentType<StyledElementType<'text'>, Theme>,
+    tspan:                    StyledComponentType<StyledElementType<'tspan'>, Theme>
+  }
+
+  declare export default StyledComponentList & {
+    [[call]]: <S : string>(S) => $ElementType<StyledComponentList, S>,
+    [[call]]: <T: Theme, Props: {}, C: React$ComponentType<Props>>(C) => StyledComponentType<React$ComponentType<{...$Exact<Props>, className?: string}>, T>
```

Comment:
Is there a better way to write this type? The `C` generic in this line was needed in order for Flow to properly infer the type of the component input to this function. Unfortunately, this means that when there's a desire to explicitly pass in `T` and `Props`, one has to write the following:

```js
const ComponentStyled = styled<ThemeType, PropsType, _>(Component)``;
```

The above example contains an unnecessary `_` for the third generic parameter. It appears that Flow doesn't allow generic defaults to be added in a libdef?? I'm not exactly sure what's going on/how to make this better. Any pointers would be greatly appreciated!